src/V4SwapRouter.sol

```diff
@@ -232,4 +245,33 @@ contract V4SwapRouter is IV4SwapRouter, BaseSwapRouter {
             )
         );
     }
+
+    /// -----------------------
+
+    /// @inheritdoc IV4SwapRouter
+    function swapWithPermit2(
+        bytes calldata data,
+        uint256 deadline,
+        ISignatureTransfer.PermitTransferFrom calldata permit,
+        bytes calldata signature
+    ) public payable virtual checkDeadline(deadline) returns (BalanceDelta) {
+        PERMIT2.permitTransferFrom(
+            permit,
+            ISignatureTransfer.SignatureTransferDetails({
+                to: address(this),
+                requestedAmount: permit.permitted.amount
+            }),
+            msg.sender,
+            signature
+        );
```

Comment:
would prefer to avoid transferring from user to swap router and then swap router to poolmanager, gonna work on tests first tho